src/pages/index.js

```diff
@@ -71,24 +119,88 @@ addCardAddButton.addEventListener("click", () => {
 
 updateProfileButton.addEventListener("click", () => {
   const userData = userInfo.getUserInfo();
-  profileHeadingInput.value = userData.userName;
-  profileDescriptionInput.value = userData.userDescription;
+  profileHeadingInput.value = userData.name;
+  profileDescriptionInput.value = userData.about;
   editProfileFormValidator.resetValidation();
   updateProfilePopup.open();
 });
 
+updateAvatarButton.addEventListener("click", () => {
+  avatarPopup.open();
+});
+
 // ! ||--------------------------------------------------------------------------------||
-// ! ||                                 Event Handlers                                 ||
+// ! ||                                 Functions                                 ||
 // ! ||--------------------------------------------------------------------------------||
+function handleValidation(form) {
+  form.enableValidation();
+}
+
 function handleAddCardFormSubmit(data) {
-  const card = renderCard(data);
-  cardSection.addItem(card);
+  this.showButtonProgress(true);
+  api
+    .addNewCard(data.name, data.link)
+    .then((res) => {
+      const card = renderCard(res);
+      cardSection.addItem(card);
+    })
+    .catch((err) => {
+      console.error(`Error: ${err}`);
+    });
+    this.showButtonProgress(false);
```

Comment:
Needs correction:  Change button state in `finally` block, otherwise the progress won't be shown (please fix that here and in all the other places)

Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally

Reply:
This has been corrected.